### PR TITLE
TE: anomaly merge code with some todos

### PR DESCRIPTION
**Pull request by puneetjaiswal**
basic anomaly merge.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyMergeExecutor.java [hunk 1]
```diff
@@ -1,72 +1,112 @@
...
+  final AnomalyFunctionDAO anomalyFunctionDAO;
+  final AnomalyMergeConfig mergeConfig;
+
+  final static Logger LOG = LoggerFactory.getLogger(AnomalyMergeExecutor.class);
 
   public AnomalyMergeExecutor(AnomalyMergedResultDAO mergedResultDAO,
-      AnomalyResultDAO anomalyResultDAO, AnomalyMergeConfig mergeConfig,
-      PinotThirdEyeClient pinotThirdEyeClient) {
+      AnomalyFunctionDAO anomalyFunctionDAO, AnomalyResultDAO anomalyResultDAO,
+      AnomalyMergeConfig mergeConfig) {
```
Comment (puneetjaiswal) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyMergeExecutor.java]: todo: mergeConfig will be saved with AnomalyFunction. keeping it temporarily here.

**Comment by kishoreg** [comment]
lgtm

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/cache/CollectionSchemaCacheLoader.java [hunk 2]
```diff
@@ -120,7 +120,7 @@ private TimeSpec fromTimeFieldSpec(TimeFieldSpec timeFieldSpec, String format) {
     TimeGranularity outputGranularity = new TimeGranularity(GRANULARITY_SIZE,
         timeFieldSpec.getOutgoingGranularitySpec().getTimeType());
     TimeSpec spec =
-        new TimeSpec(timeFieldSpec.getOutGoingTimeColumnName(), outputGranularity, format);
+        new TimeSpec(timeFieldSpec.getOutgoingTimeColumnName(), outputGranularity, format);
```
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/cache/CollectionSchemaCacheLoader.java]: Why was this changed back? This is breaking the build again. This was fixed correctly
